dev-docs/RFCs/v6.1/json-layers-rfc.md

```diff
@@ -0,0 +1,105 @@
+# RFC: JSON Layers
+
+* **Authors**: Ib Green
+* **Date**: July 2018
+* **Status**: Draft Proposal
+
+
+## Summary
+
+This RFC proposes adding adaptor classes that accept JSON descriptions of views and layers.
+
+
+## Background
+
+Especially in the infovis space, there is a growing need to be able to generate powerful visualizations directly from the backend. Being able to describe a visualization in abstract terms and send it to the front-end for display withouth having knowledge about how to code front-end applications can be valuable.
```

Comment:
> Especially in the infovis space, there is a growing need to be able to generate powerful visualizations directly from the backend.

For statistical charts and well-received visualizations such as node-link diagrams, maybe. Basically, we need to know exactly how users want to map their data to the visual channels ahead of time and hardcode the mapping for them in the JSONLayer (unless we can encode the layout algorithms as JSON strings). 

I see the value of prescribed backend-generated visualizations when there's a need for real-time streaming, but I doubt JSON is the right format for that.

Regarding declarative APIs for InfoVis, there's already good [studies](https://vega.github.io/vega-lite/examples/) on which types of visualizations can be defined declaratively. I agree we should keep our declarative interface as close as possible to other incarnations of ours, but we need to design carefully not to reject users switching from the vega community.

I guess we deprioritized the declarative API last time due to we could not find a good user cohort for it. Or, who will use this JSONLayer? For those who want to get their hand dirty, they can just code in JS and use our layers directly. For users who just want the capability, we build workflow tools for them. If we can identify good use cases for the backend generated InfoVis (like streaming pre-layout node-link graph from backend via Apache Arrow), I'm totally for it.